index.ios.js

```diff
@@ -224,23 +207,67 @@ export async function openInbox(options = { compose = false }) {
     app = await askAppChoice(title, message, cancelLabel, removeText);
   }
 
-  let url = null;
-  switch (app) {
-    default:
-      url = prefixes[app];
+  if (!app) {
+    throw new EmailException('No email app found');
   }
 
-  let params = '';
-  if(option.compose || options.to || options.subject || options.body || options.cc || options.bcc) {
-    params = getUrlParams(app, options);
+  return app;
+}
 
-    if (app === 'apple-mail') {
-      // apple mail base url to compose an email is mailto
-      url = 'mailto:';
-    }
+/**
+ * Open an email app, or let the user choose what app to open.
+ *
+ * @param {{
+  *     app: string | undefined | null,
+  *     title: string,
+  *     message: string,
+  *     cancelLabel: string,
+  *     removeText: boolean
+  * }} options
+  */
+export async function openInbox(options) {
+  if (!options || typeof options !== "object") {
+    throw new EmailException(
```

Comment:
I think we should throw this exception only if options exists and it is not an object. That way developers would be able to call `openInbox()` instead of `openInbox({})`.

Reply:
Agreed!